File: PickerView/app/src/main/java/com/example/pickerview/InitializeCalculator.java
package com.example.pickerview;


import android.widget.NumberPicker;
import android.widget.TextView;

/**
 * Created by HimelR on 18-Jan-18.
 */

class InitializeCalculator {

    private final TextView textView;
    private final Person person;
    private final NumberPicker numberPicker;
    private final NumberPicker numberPicker2;

    InitializeCalculator(TextView textView, NumberPicker numberPicker, NumberPicker numberPicker2){
        this.textView = textView;
        this.numberPicker = numberPicker;
        this.numberPicker2 = numberPicker2;
        person = new Person(170, 70);
        setValues();
    }

    private void setValues(){
        getNumberPicker().setMinValue(30);
        getNumberPicker().setMaxValue(130);
        getNumberPicker2().setMinValue(50);
        getNumberPicker2().setMaxValue(210);
        getNumberPicker().setValue(70);
        getNumberPicker2().setValue(170);
        this.textView.setText("Value:\n" + person.calculateBmi());
    }

    NumberPicker getNumberPicker() {
        return numberPicker;
    }

    NumberPicker getNumberPicker2() {
        return numberPicker2;
    }

    void calculate(int value, boolean s){
        if(s){
            person.setWeight(value);
        }
        else {
            person.setHeight(value);
        }
        this.textView.setText("Value:\n" + person.calculateBmi());


    }
}

File: BMIKotlin/app/src/main/java/com/example/bmikotlin/InitializeCalculator.kt
package com.example.bmikotlin

import android.widget.NumberPicker
import android.widget.TextView

/**
 * Created by HimelR on 07-Feb-18.
 */
class InitializeCalculator(private val text1:TextView, val numberp:NumberPicker, val numberp2:NumberPicker,private val person: Person ) {
    init {
        setup()
    }

    private fun setup() {
       numberp.minValue = 30
       numberp.maxValue = 130
       numberp2.minValue = 120
       numberp2.maxValue = 210
       numberp.value = 70
       numberp2.value = 170
       text1.text = ("Value:\n" + person.bmi())
    }

    fun calculate(value: Int, s: Boolean) {
        if (s) {
            person.weight = value
        } else {
            person.height = value
        }
        text1.text = ("Value:\n" + person.bmi())


    }
}